nativeauthenticator/nativeauthenticator.py

```diff
@@ -33,20 +46,39 @@ def authenticate(self, handler, data):
         if user.is_authorized and user.is_valid_password(data['password']):
             return data['username']
 
-    def get_or_create_user(self, username, password):
-        user = User.find(self.db, username)
-        if not user:
-            user = User(name=username, admin=False)
-            self.db.add(user)
+    def is_password_common(self, password):
+        common_credentials_file = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            'common-credentials.txt'
+        )
+        if not self.COMMON_PASSWORDS:
+            with open(common_credentials_file) as f:
+                self.COMMON_PASSWORDS = f.read().splitlines()
+        return password.lower() in self.COMMON_PASSWORDS
+
+    def is_password_strong(self, password):
+        checks = [
```

Comment:
I think these should be two config options:

1. `disallow_common_passwords` - Checks if password used is common. Default to True
2. `minimum_password_length` - Only allow passwords longer than this. Can be set to 0 to disable length checking. Defaults to 8.

Also, when checking a constant number of variables like this, it's clearer to just use a boolean expression directly `len(password) >= self.minimum_password_length and not self.is_common_password(password)` than using `all`